### PR TITLE
clean up old console resources

### PR DESCRIPTION
**Pull request by ray-harris**
Signed-off-by: Ray Harris <raharris@redhat.com>

### CODE REVIEW

**Review by JakobGray — CHANGES_REQUESTED** [review]
We are applying the `hive-clusterimagesets-subscription-fast-0` Subscription as part of the console component ([ref](https://github.com/stolostron/multiclusterhub-operator/blob/main/pkg/templates/charts/toggle/console/templates/subscription-fast.yaml)). When testing did this lead to an infinite loop as the operator creates and deletes this resource? 

There exists a dedicated uninstall.go file in this same package, along with an `ensureRemovalsGone` function which is called in our reconciler [here](https://github.com/stolostron/multiclusterhub-operator/blob/main/controllers/multiclusterhub_controller.go#L475). It would be better to situate cleanup there to avoid duplication.

But rather than cleaning up the resource it may be enough to simply not apply it in the first place and verify it is not present on upgrade. Testing out an upgrade can confirm this.

I believe this removal is based on DangPeng's comment in https://github.com/stolostron/backlog/issues/26603#issuecomment-1316555778. If that subscription is no longer used then chances are the [channel](https://github.com/stolostron/multiclusterhub-operator/blob/main/pkg/templates/charts/toggle/console/templates/channel.yaml) it uses also is no longer needed. That would be a good thing to confirm with DangPeng.

**Comment by ray-harris** [comment]
@JakobGray I haven't completed testing. Working on adding a test case for an existing appsub which should show if it get stuck in a loop. But I didn't realize we were applying the appsub. I'll check to see if the channel doesn't need it if we can just not apply it and then see if it isn't present.

File: controllers/uninstall.go [hunk 1]
```diff
@@ -82,6 +82,14 @@ var (
 				types.NamespacedName{Name: "management-ingress-sub", Namespace: m.Namespace},
 				schema.GroupVersionKind{Group: "apps.open-cluster-management.io", Kind: "Subscription", Version: "v1"},
 			),
+			newUnstructured(
```
Comment (JakobGray) [inline controllers/uninstall.go:85]: Yesterday you commented
> I'll check to see if the channel doesn't need it if we can just not apply it and then see if it isn't present.

Were you able to test that? I kind of imagined these resources would not need to be explicitly uninstalled, for the same reason that none of the others do.

Comment (JakobGray) [reply controllers/uninstall.go:85]: @ray-harris Any update?

Comment (ray-harris) [reply controllers/uninstall.go]: the resources are removed when taken out of the templates. removing these lines.

**Review by JakobGray — APPROVED** [review]

**Pull request merged**